File: TestProject/SinglePackageRepoTest.cs
using Brilliantech.Packaging.WS;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System;
using Brilliantech.Packaging.Data;
using System.Linq;

namespace TestProject
{
    
    
    /// <summary>
    ///这是 SinglePackageRepoTest 的测试类，旨在
    ///包含所有 SinglePackageRepoTest 单元测试
    ///</summary>
    [TestClass()]
    public class SinglePackageRepoTest
    {


        private TestContext testContextInstance;

        /// <summary>
        ///获取或设置测试上下文，上下文提供
        ///有关当前测试运行及其功能的信息。
        ///</summary>
        public TestContext TestContext
        {
            get
            {
                return testContextInstance;
            }
            set
            {
                testContextInstance = value;
            }
        }

        #region 附加测试特性
        // 
        //编写测试时，还可使用以下特性:
        //
        //使用 ClassInitialize 在运行类中的第一个测试前先运行代码
        //[ClassInitialize()]
        //public static void MyClassInitialize(TestContext testContext)
        //{
        //}
        //
        //使用 ClassCleanup 在运行完类中的所有测试后再运行代码
        //[ClassCleanup()]
        //public static void MyClassCleanup()
        //{
        //}
        //
        //使用 TestInitialize 在运行每个测试前先运行代码
        //[TestInitialize()]
        //public void MyTestInitialize()
        //{
        //}
        //
        //使用 TestCleanup 在运行完每个测试后运行代码
        //[TestCleanup()]
        //public void MyTestCleanup()
        //{
        //}
        //
        #endregion


        /// <summary>
        ///FindSinglePackage 的测试
        ///</summary>
        [TestMethod()]
        public void FindSinglePackageTest()
        {
            PackagingDataDataContext context = new PackagingDataDataContext("D:\\vm\\DBBackup\\Minor-TSK-055\\package.sdf"); // TODO: 初始化为适当的值
            SinglePackageRepo target = new SinglePackageRepo(context); // TODO: 初始化为适当的值
            string pid = "P120825085853069"; // TODO: 初始化为适当的值
            string wrkstnr = string.Empty; // TODO: 初始化为适当的值
            string tnr = string.Empty ; // TODO: 初始化为适当的值
            string partnr = string.Empty; // TODO: 初始化为适当的值
            int status =-1; // TODO: 初始化为适当的值
            DateTime fromDate = new DateTime(1990,1,1); // TODO: 初始化为适当的值
            DateTime toDate = new DateTime(2100,1,1); // TODO: 初始化为适当的值
            System.Collections.Generic.List<FullPackageInfo > actual;
            actual = target.FindSinglePackage(pid, wrkstnr, tnr, partnr, status, fromDate, toDate);
            Assert.IsNotNull (actual);
            Assert.AreEqual(actual.Count(), 1);
         
        }

        /// <summary>
        ///GetItemsByPId 的测试
        ///</summary>
        [TestMethod()]
        public void GetItemsByPIdTest()
        {
            PackagingDataDataContext context = new PackagingDataDataContext("D:\\vm\\DBBackup\\Minor-TSK-055\\package.sdf"); // TODO: 初始化为适当的值
            SinglePackageRepo target = new SinglePackageRepo(context); // TODO: 初始化为适当的值
            string pid = "P120813083031247"; // TODO: 初始化为适当的值
            IQueryable<PackageItem> actual;
            actual = target.GetItemsByPId(pid);
            Assert.IsNotNull (actual);
            Assert.AreEqual (actual.ToList().Count, 18);
        }
    }
}

File: TestProjectProcess/SinglePackageRepoTest.vb
Imports Brilliantech.Packaging.Data

Imports Microsoft.VisualStudio.TestTools.UnitTesting

Imports Brilliantech.Packaging.WS



'''<summary>
'''This is a test class for SinglePackageRepoTest and is intended
'''to contain all SinglePackageRepoTest Unit Tests
'''</summary>
<TestClass()> _
Public Class SinglePackageRepoTest


    Private testContextInstance As TestContext

    '''<summary>
    '''Gets or sets the test context which provides
    '''information about and functionality for the current test run.
    '''</summary>
    Public Property TestContext() As TestContext
        Get
            Return testContextInstance
        End Get
        Set(value As TestContext)
            testContextInstance = Value
        End Set
    End Property

#Region "Additional test attributes"
    '
    'You can use the following additional attributes as you write your tests:
    '
    'Use ClassInitialize to run code before running the first test in the class
    '<ClassInitialize()>  _
    'Public Shared Sub MyClassInitialize(ByVal testContext As TestContext)
    'End Sub
    '
    'Use ClassCleanup to run code after all tests in a class have run
    '<ClassCleanup()>  _
    'Public Shared Sub MyClassCleanup()
    'End Sub
    '
    'Use TestInitialize to run code before running each test
    '<TestInitialize()>  _
    'Public Sub MyTestInitialize()
    'End Sub
    '
    'Use TestCleanup to run code after each test has run
    '<TestCleanup()>  _
    'Public Sub MyTestCleanup()
    'End Sub
    '
#End Region


    '''<summary>
    '''A test for GetByItemId
    '''</summary>
    <TestMethod()> _
    Public Sub GetByItemIdTest()
        Dim context As PackagingDataDataContext = New PackagingDataDataContext("data source=E:\Project\莱尼包装\Package代码\Brilliantech.PackagingWS\Brilliantech.PackagingWS\data\package.sdf") ' TODO: Initialize to an appropriate value
        Dim target As SinglePackageRepo = New SinglePackageRepo(context) ' TODO: Initialize to an appropriate value
        Dim itemId As String = "91G207402" ' TODO: Initialize to an appropriate value
        Dim expected As SinglePackage = Nothing ' TODO: Initialize to an appropriate value
        Dim actual As SinglePackage
        actual = target.GetByItemId(itemId)
        Assert.IsNotNull(actual)

    End Sub


    <TestMethod()> _
    Public Sub TestModify()

    End Sub

    '''<summary>
    '''Modify 的测试
    '''</summary>
    <TestMethod()> _
    Public Sub ModifyTest()
        Dim context As PackagingDataDataContext = New PackagingDataDataContext("Data Source=192.168.0.253\dev001;Initial Catalog=Leoni_Packaging_prod;User ID=sa;Password=123456@") ' TODO: 初始化为适当的值
        Dim target As IBaseRepo(Of SinglePackage) = New SinglePackageRepo(context) ' TODO: 初始化为适当的值
        Dim entity As SinglePackage = New SinglePackage With {.capa = 46, .containerID = "MBSTP", .packageID = "TMB001", .partNr = "91G004702", .planedDate = Now, .rowguid = New Guid("{81f9b732-2e87-e211-93f2-000c2901ecf9}"), .status = 99, .wrkstnID = "Minor-TSK-041"}
        Try
            target.Modify(entity)
            Assert.Fail()
        Catch ex As Exception

        End Try


    End Sub
End Class
